crates/sortition/src/sortition.rs

```diff
@@ -25,84 +28,243 @@ pub struct GetHasNode {
+        match self {
+            SortitionBackend::Score { registered } => registered,
+            SortitionBackend::Distance { .. } => {
+                panic!("attempted to mutate Score registered on Distance backend")
+            }
+        }
+    }
+
+    fn nodes_view(&self) -> Vec<String> {
+        match self {
+            SortitionBackend::Distance { nodes } => nodes.iter().cloned().collect(),
+            SortitionBackend::Score { registered } => {
+                registered.iter().map(|n| n.address.to_string()).collect()
+            }
         }
+    }
 
-        let registered_nodes: Vec<Address> = self
-            .nodes
-            .clone()
-            .into_iter()
-            // TODO: error handling
-            .map(|b| b.parse().unwrap())
-            .collect();
-
-        let Ok(committee) =
-            DistanceSortition::new(seed.into(), registered_nodes, size).get_committee()
-        else {
-            return Err(anyhow!("Could not get committee!"));
+    fn contains_distance(&self, seed: Seed, size: usize, address: &str) -> Result<bool> {
+        let nodes = match self {
+            SortitionBackend::Distance { nodes } => nodes,
+            _ => return Err(anyhow!("wrong backend for distance contains")),
         };
 
+        if nodes.is_empty() || size == 0 {
+            return Ok(false);
+        }
+
+        let registered_nodes: Vec<Address> = nodes
+            .iter()
+            .cloned()
+            .map(|b| b.parse::<Address>())
+            .collect::<Result<_, _>>()?;
+
+        let committee =
+            DistanceSortition::new(seed.into(), registered_nodes, size).get_committee()?;
+
         Ok(committee
             .iter()
             .any(|(_, addr)| addr.to_string() == address))
     }
 
+    fn contains_score(&self, seed: Seed, size: usize, address: &str) -> Result<bool> {
+        let registered = match self {
+            SortitionBackend::Score { registered } => registered,
+            _ => return Err(anyhow!("wrong backend for score contains")),
+        };
+
+        if registered.is_empty() || size == 0 {
+            return Ok(false);
+        }
+
+        let committee = ScoreSortition::new(size).get_committee(seed.into(), registered)?;
+        let want: Address = address.parse()?;
+        Ok(committee.iter().any(|w| w.address == want))
+    }
+}
+
+/// Per-chain module: stores the chosen algorithm and its backend storage.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct SortitionModule {
+    sortition: SortitionType,
+    backend: SortitionBackend,
+}
+
+impl Default for SortitionModule {
+    fn default() -> Self {
+        Self {
+            sortition: SortitionType::Distance,
+            backend: SortitionBackend::default(),
+        }
+    }
+}
+
+impl SortitionModule {
+    /// Create a module with a specific algorithm.
+    pub fn new(sortition: SortitionType) -> Self {
+        let backend = match sortition {
```

Comment:
If we specify the backend type in the configuration would it be easier to just expect an enumerated backend in construction? 

thinking:

```rust
pub fn new(backend: SortitionBackend) -> Self;
```

```rust
#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct SortitionModule {
    backend: SortitionBackend,
}
```

Reply:
related to above